From b086abb953c7a8853a4a27615230ab65e0f35f83 Mon Sep 17 00:00:00 2001
From: Darren Huang <darren.huang@advantech.com.tw>
Date: Mon, 4 Nov 2024 17:43:54 +0800
Subject: [PATCH 2/3] qps615: auto load module

---
 tc956x_pci.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tc956x_pci.c b/tc956x_pci.c
index af65270..2d39523 100755
--- a/tc956x_pci.c
+++ b/tc956x_pci.c
@@ -4314,6 +4314,7 @@ static const struct pci_device_id tc956xmac_id_table[] = {
 #endif
 	{}
 };
+MODULE_DEVICE_TABLE(pci, tc956xmac_id_table);
 
 static SIMPLE_DEV_PM_OPS(tc956xmac_pm_ops, tc956x_pcie_suspend, tc956x_pcie_resume);
 
-- 
2.25.1

